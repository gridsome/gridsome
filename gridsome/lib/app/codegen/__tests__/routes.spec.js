const App = require('../../App')
const genRoutes = require('../routes')
const { NOT_FOUND_NAME } = require('../../../utils/constants')

test('generate a simple route', async () => {
  const app = await new App(__dirname).init()

  app.pages.createPage({
    path: '/page',
    component: './__fixtures__/PageA.vue'
  })

  expect(genRoutes(app)).toMatchSnapshot()
})

test('generate a route with meta', async () => {
  const app = await new App(__dirname).init()

  app.pages.createPage({
    path: '/page',
    component: './__fixtures__/PageA.vue',
    route: {
      meta: {
        string: 'test',
        boolean: true,
        number: 2
      }
    }
  })

  expect(genRoutes(app)).toMatchSnapshot()
})

test('generate a route with raw code as meta', async () => {
  const app = await new App(__dirname).init()

  app.pages.createPage({
    path: '/page',
    component: './__fixtures__/PageA.vue',
    route: {
      meta: {
        $log: `() => console.log('hello!')`
      }
    }
  })

  expect(genRoutes(app)).toMatchSnapshot()
})

test('generate component imports as variables', async () => {
  const app = await new App(__dirname).init()

  app.pages.createPage({
    path: '/404',
    name: NOT_FOUND_NAME,
    component: './__fixtures__/PageC.vue'
  })

  for (let i = 1; i < 4; i++) {
    app.pages.createPage({
      path: '/page-a/' + i,
      component: './__fixtures__/PageA.vue'
    })
  }

  app.pages.createPage({
    path: '/page-b',
    component: './__fixtures__/PageB.vue'
  })

  app.pages.createPage({
    path: '/user/:id',
    component: './__fixtures__/PageC.vue'
  })

  expect(genRoutes(app)).toMatchSnapshot()
})


test('generate redirect routes', async () => {
  const app = await new App(__dirname, {
    config: {
      redirects: [
        { from: '/old', to: '/page' }
      ]
    }
  }).init()

  app.pages.createPage({
    path: '/page',
    component: './__fixtures__/PageA.vue'
  })

  expect(genRoutes(app)).toMatchSnapshot()
})
