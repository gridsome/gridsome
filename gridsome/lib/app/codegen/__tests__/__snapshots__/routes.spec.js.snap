// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate a route with meta 1`] = `
"const c1 = () => import(/* webpackChunkName: \\"page--fixtures--page-a-vue\\" */ \\"__fixtures__/PageA.vue\\")

export default [
  {
    path: \\"/page/\\",
    component: c1,
    meta: {
      string: \\"test\\",
      boolean: true,
      number: 2
    }
  }
]
"
`;

exports[`generate a route with raw code as meta 1`] = `
"const c1 = () => import(/* webpackChunkName: \\"page--fixtures--page-a-vue\\" */ \\"__fixtures__/PageA.vue\\")

export default [
  {
    path: \\"/page/\\",
    component: c1,
    meta: {
      $log: () => console.log('hello!')
    }
  }
]
"
`;

exports[`generate a simple route 1`] = `
"const c1 = () => import(/* webpackChunkName: \\"page--fixtures--page-a-vue\\" */ \\"__fixtures__/PageA.vue\\")

export default [
  {
    path: \\"/page/\\",
    component: c1
  }
]
"
`;

exports[`generate component imports as variables 1`] = `
"const c1 = () => import(/* webpackChunkName: \\"page--fixtures--page-a-vue\\" */ \\"__fixtures__/PageA.vue\\")
const c2 = () => import(/* webpackChunkName: \\"page--fixtures--page-c-vue\\" */ \\"__fixtures__/PageC.vue\\")
const c3 = () => import(/* webpackChunkName: \\"page--fixtures--page-b-vue\\" */ \\"__fixtures__/PageB.vue\\")

export default [
  {
    path: \\"/page-a/1/\\",
    component: c1
  },
  {
    path: \\"/page-a/2/\\",
    component: c1
  },
  {
    path: \\"/page-a/3/\\",
    component: c1
  },
  {
    name: \\"__user_id\\",
    path: \\"/user/:id\\",
    component: c2,
    meta: {
      dataPath: \\"/user/_id.json\\",
      dynamic: true
    }
  },
  {
    path: \\"/page-b/\\",
    component: c3
  },
  {
    name: \\"404\\",
    path: \\"*\\",
    component: c2,
    meta: {
      dataPath: \\"/404.json\\"
    }
  }
]
"
`;

exports[`generate redirect routes 1`] = `
"const c1 = () => import(/* webpackChunkName: \\"page--fixtures--page-a-vue\\" */ \\"__fixtures__/PageA.vue\\")

export default [
  {
    path: \\"/old\\",
    redirect: \\"/page\\"
  },
  {
    path: \\"/page/\\",
    component: c1
  }
]
"
`;
